modules/custom/az_finder/src/Form/AZFinderSettingsForm.php

```diff
@@ -0,0 +1,412 @@
+      $container->get('az_finder.vocabulary'),
+      $container->get('az_finder.overrides'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'az_finder_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['az_finder.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $form['#tree'] = TRUE;
+
+    // Page description.
+    $form['description'] = [
+      '#type' => 'item',
+      '#markup' => $this->t('
+        <p>Manage the settings you would like to use with exposed AZ Finder forms.</p>
+        <p>For more information about the Quickstart Finder, please <a href="https://quickstart.arizona.edu/create-content/quickstart-finder">visit the Quickstart website</a>.</p>
+      '),
+    ];
+
+    // How these settings work section.
+    $form['how_it_works'] = [
+      '#type' => 'details',
+      '#title' => $this->t('How do these settings work?'),
+      '#open' => FALSE,
+      '#description' => $this->t('
+        <p>The default settings are applied to all taxonomy vocabularies as a starting point.</p>
+        <p>Each Finder view display can have custom overrides to expand or collapse specific sections by default.</p>
+      '),
+    ];
+
+    // Filter Widget Settings section.
+    $form['az_finder_tid_widget'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Filter Widget Settings'),
+      '#open' => TRUE,
+    ];
+
+    // Default state select field.
+    $form['az_finder_tid_widget']['default_state'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Default Display'),
+      '#description' => $this->t('Choose how taxonomy terms with children should behave by default everywhere.<br />These settings are not context aware, so if you choose collapsed, your term must be using a collapsible element for this to work.'),
+      '#weight' => -2,
+      '#options' => [
+        'expand' => $this->t('Expanded'),
+        'collapse' => $this->t('Collapsed'),
+      ],
+      '#config_target' => 'az_finder.settings:tid_widget.default_state',
+    ];
+
+    // Fetch existing overrides from the AZFinderOverrides service.
+    $config_overrides = $this->azFinderOverrides->getExistingOverrides();
+
+    // Get current overrides from form state.
+    $session_overrides = $form_state->getValue(['az_finder_tid_widget', 'overrides']) ?? [];
+
+    // Normalize session overrides structure if needed.
+    $normalized_session_overrides = [];
+    foreach ($session_overrides as $key => $override) {
+      if (!isset($override['origin'])) {
+        $override['origin'] = 'session';
+      }
+      $normalized_session_overrides[$key] = $override;
+    }
+
+    // Combine overrides.
+    $overrides = array_merge($config_overrides, $normalized_session_overrides);
+
```

Comment:
This is where the overrides are merged together when the form is being interacted with.

`$config_overrides` are from active configuration
`$normalized_session_overrides` are from the current state of the form


These are added to the $form [like this](https://github.com/az-digital/az_quickstart/pull/3420/files#diff-3afdb43ec0cded9e8f1617d4398eed2556d3f366236b7301497eddc6bc5cd00eR218-R221)
```
    // Add override sections.
    foreach ($overrides as $override) {
      $this->addOverrideSection($form, $form_state, $override);
    }
 ```

`$form['az_finder_tid_widget']['overrides']` contains a bunch of stuff that isn't overrides. Like those form elements you keep trying to move around., you might try sorting here.

Reply:
If you were to `dpm($overrides);` at this point, I believe you will also see the missing vocabulary on session overrides, so it seems that the [function here](https://github.com/az-digital/az_quickstart/pull/3420/files#diff-3afdb43ec0cded9e8f1617d4398eed2556d3f366236b7301497eddc6bc5cd00eR352-R360) is not updating form state in time to be included in the `$normalized_session_overrides` array.

Reply:
I added some UI steps here to see the issue.
https://github.com/az-digital/az_quickstart/pull/3420#:~:text=Issue%20Recreation%20Steps